tests/pytest_tests/system_tests/tests_launch/test_launch.py

```diff
@@ -96,3 +98,33 @@ def test_launch_incorrect_backend(
 
         assert "Resource name not among available resources" in str(e_info)
         r.finish()
+
+
+def test_launch_multi_run(relay_server, runner, user, wandb_init, test_settings):
+    with runner.isolated_filesystem(), mock.patch.dict(
+        "os.environ", {"WANDB_RUN_ID": "test", "WANDB_LAUNCH": "true"}
+    ):
+        run1 = wandb_init()
+        run1.finish()
+
+        run2 = wandb_init()
+        run2.finish()
+
+        assert run1.id == "test"
+        assert run2.id != "test"
+
+
+def test_launch_multi_run_context(
+    relay_server, runner, user, wandb_init, test_settings
+):
+    with runner.isolated_filesystem(), mock.patch.dict(
+        "os.environ", {"WANDB_RUN_ID": "test", "WANDB_LAUNCH": "true"}
+    ):
+        with wandb_init() as run1:
+            run1.log({"test": 1})
+
+        with wandb_init() as run2:
+            run2.log({"test": 2})
+
+        assert run1.id == "test"
+        assert run2.id != "test"
```

Comment:
So when exactly would such a situation occur? If you were the user, what would you expect? would you care at all about the run id's of the subsequent runs? Would e.g. prefixing the run_ids for the later runs with the original RUN_ID (or some other name-spacing) be more appropriate?

Reply:
This is probably the most expected behavior for the largest set of users, since if they are calling wandb init several times in the same run they don't want them to have the same run id. I think in the future, we'll try to do more to track the structure of runs created and decide on runids, run groups etc. using the structure of runs created by a single entrypoint